corehq/apps/users/static/users/js/mobile_workers.js

```diff
@@ -0,0 +1,121 @@
+hqDefine("users/js/mobile_workers", function () {
+    var mobileWorker = function (options) {
+        var self = ko.mapping.fromJS(options);
+
+        self.mark_activated = ko.observable(false);
+        self.mark_deactivated = ko.observable(false);
+        self.action_error = ko.observable('');
+
+        self.showActivate = ko.computed(function () {
```

Comment:
I think you could remove a lot of this logic.
1. don't send `action` from the backend (instead, I would send `status` or more preferably `is_active`, so it matches the django model)
2. don't compute showActivate or showDeactivate
3. in the template, instead of `visible: showActivate`, it might be clearer to do `status === deactivated` or `!isActive`. That way the action is clearly linked to the status of the mobile worker. 
4. Then, in your `modifyStatus` function, you don't need to have the cryptic boolean argument (since `isActive` will be part of the view model). 

you might also then be able to remove the `mark_deactivated` and `mark_activated` fields, but I think I might be missing the nuance of why those exist.